Runtime/utils/ABI.cs

```diff
@@ -86,6 +86,48 @@ private static PreparedParam GetEncodeValue(object obj, string inputType)
                 {
                     value = new BigInteger(intValue);
                 }
+                else if (obj is BigInteger bInt)
+                {
+                    value = bInt;
+                }
+                else
+                {
+                    throw new System.NotImplementedException();
+                }
+                if (size % 32 != 0)
+                {
+                    size = (int)(Math.Ceiling(size / 32f) * 32);
+                }
+                return EncodeNumber(value, signed, size);
+            }
+            else if (inputType == "bool")
+            {
+                bool signed = false;
+                int size = 1;
+                BigInteger value;
```

Comment:
[SHOULD] Extract method if logic is repeated.

Reply:
The logic is stable, I will not change!

Reply:
Done!